src/config/dnsmasq_config.c

```diff
@@ -547,6 +552,14 @@ bool __attribute__((nonnull(1,3))) write_dnsmasq_config(struct config *conf, boo
 		fputs("local=//\n\n", pihole_conf);
 	}
 
+	// Ensure that home.arpa domains (RFC 8375) are not sent upstream, unless configured by
+	// user as local domain, with server setting applied above
+	if (domain_homearpa == false)
```

Comment:
```suggestion
	if (!domain_homearpa)
```
just for code style uniformity